dm/worker/worker.go

```diff
@@ -306,6 +304,110 @@ func (w *Worker) QueryError(name string) []*pb.SubTaskError {
 	return w.Error(name)
 }
 
+// purgeRelayDir will clear all contents under w.cfg.RelayDir
+func (w *Worker) purgeRelayDir() error {
+	if !w.cfg.EnableRelay {
+		return nil
+	}
+	dir := w.cfg.RelayDir
+	d, err := os.Open(dir)
+	// fail to open dir, return directly
+	if err != nil {
+		return nil
```

Comment:
only when err is `no such file or directory` can return nil?

Reply:
@lichunzhu PTAL

Reply:
Revised in https://github.com/pingcap/dm/pull/473/commits/2783ba63bdef5122af8ba5ff5d8d46afc66e0a46